vnet/lib/vnet/endpoints/1.0/interfaces.rb

```diff
@@ -128,33 +128,43 @@ def self.port_delete_post_shared_params
   # Segments:
   #
 
-  param_uuid M::Interface
-  param_uuid M::Segment, :segment_uuid
-  param :static, :Boolean
-  put '/:uuid/segments/:segment_uuid' do
-    # TODO: Move the 'uuid_to_id' calls to nodeapi and add a
-    # base_foobar module that let's us easily extract the foo_id from
-    # uuid, id or nil according to the requirements.
-    interface = uuid_to_id(M::Interface, 'uuid', 'interface_id')
-    segment = uuid_to_id(M::Segment, 'segment_uuid', 'segment_id')
-
-    # TODO: Use the default update handling code instead.
-
-    if params.has_key?('static')
-      if params['static']
-        result = M::InterfaceSegment.set_static(interface.id, segment.id)
+  ASSOCS = [
+    [M::Network, M::InterfaceNetwork, R::InterfaceNetwork, :network],
+    [M::Segment, M::InterfaceSegment, R::InterfaceSegment, :segment],
+    [M::RouteLink, M::InterfaceRouteLink, R::InterfaceRouteLink, :route_link]
+  ].freeze
```

Comment:
This adds a bunch of relations to the interfaces endpoints but you haven't added matching tests yet in https://github.com/axsh/openvnet/blob/develop/vnet/spec/vnet/endpoints/1.0/interfaces_spec.rb. Should be simple since it's just another call to some shared examples that already exist.